### ISSUE TITLE
GM Level Permission Bugs

### ISSUE THREAD

**Issue by AbuGosok** [rathena/rathena#511]
item_trade.txt
// Legend for 'TradeMask' field (bitmask):
// 1   - item can't be dropped
// 2   - item can't be traded (nor vended)
// 4   - wedded partner can override restriction 2
// 8   - item can't be sold to npcs
// 16  - item can't be placed in the cart
// 32  - item can't be placed in the storage
// 64  - item can't be placed in the guild storage
// 128 - item can't be attached to mail
// 256 - item can't be auctioned
//
// Example:
// 1161,67,50
//	Balmung cannot be dropped, traded, or placed in Guild Storage (1+2+64).
//	Only groups of group-level 50 and up can override this setting.

501,475,5
It means item id 501 is 1+2+8+16+64+128+256 (can't be dropped, can be placed in storage) and only group-level 5 and up can override this setting.

I created GM level 99, but I can not override it (can't be dropped, can be placed in storage). GM level 10 and 99 should be able to override this, right?

2nd, about mapflag permission:
prt_fild05	mapflag	nocommand	5

It means that player can't type any atcommand in prt_fild05, but GM level 5, 10 and 99 can override it, right?

But in reality, only GM level 99 able to type atcommand. GM level 5 and 10 still denied on that particular map.

**Comment by reigneil** [comment]
i can confirm this... maybe because they change the way permission is applied now.

**Comment by aleos89** [comment]
Is the group level above the restriction level? Just making sure you're not confusing Group ID with Group Level.

**Comment by aleos89** [comment]
Finally got some time to trace this down. It fails to drop the item in pc.c::pc_dropitem()
```
	if (!map_addflooritem(&sd->status.inventory[n], amount, sd->bl.m, sd->bl.x, sd->bl.y, 0, 0, 0, 2, 0))
		return false;
```

Here we pass flag 2. The description of the flags for map_addflooritem() are:
@param flag: &1 MVP item. &2 do stacking check. &4 bypass droppable check.

This check was implemented in bb6c9b6 and 0b94fb3. I'm not sure if we should add the flag 4 in there or not. Maybe @Euphy or @lighta can give some more details.
